### PR TITLE
[#10032] feat(IT): Introduce docker-compose support for DorisContainer

### PR DESCRIPTION
**Pull request by zhoukangcn**
### What changes were proposed in this pull request?

Introduce docker-compose based Doris container environment using separate FE and BE containers with official apache/doris:2.1.7 images, enabling proper multi-node cluster testing and Doris 2.1.x compatibility.

### Why are the changes needed?

Fix: #10032

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

IT

### CODE REVIEW

**Comment by zhoukangcn** [comment]
@yuqi1129 Could you help review this PR.

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java [hunk 1]
```diff
@@ -63,78 +84,130 @@ protected DorisContainer(
       Map<String, String> envVars,
       Optional<Network> network) {
     super(image, hostName, ports, extraHosts, filesToMount, envVars, network);
-  }
```
[the remark this answers is not in the document]
Comment (zhoukangcn) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java:90]: I'd suggest not making this change for now. The current implementation works fine and removing the BaseContainer extension would be a bigger refactor that's out of scope for this PR. We can address it separately later if needed.

@yuqi1129 What do you think?

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java [hunk 2]
```diff
@@ -63,78 +84,136 @@ protected DorisContainer(
...
                     if (alive.equalsIgnoreCase("true") && totalCapacityFloat > 0.0f) {
-                      LOG.info("Doris container startup success!");
-                      return true;
+                      aliveCount++;
                     }
                   }
                 }
-                LOG.info("Doris container is not ready yet!");
+
+                if (aliveCount >= BE_SCALE) {
```
Comment (yuqi1129) [inline integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java:212]: Will it happen that `aliveCount` is larger than `BE_SCALE`?

Comment (Kangs-Claw) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java:212]: Good question! In theory, `aliveCount` should not exceed `BE_SCALE` since we only start 3 BE containers via `withScaledService(BE_SERVICE, BE_SCALE)`. However, using `>=` instead of `==` is more robust because:

1. It handles edge cases where extra backends might be present from previous test runs
2. It's more defensive programming - if somehow more backends appear, the check still passes
3. It aligns with the intent: "we need at least BE_SCALE backends alive"

So while it's unlikely to happen in normal circumstances, the `>=` check is safer and more correct.

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (yuqi1129) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java:90]: Yeah, we can use another PR to resolve this problem. I will appreciate it if you open an issue to track it.

[the remark this answers is not in the document]
Comment (zhoukangcn) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java:90]: done

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java (hunk 2 above)
Comment (zhoukangcn) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java:212]: @yuqi1129 Do you have more suggestion?

Comment (yuqi1129) [reply integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java:212]: No, it's also fine with me.

File: integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java [hunk 3]
```diff
@@ -63,78 +84,136 @@ protected DorisContainer(
...
+    composeContainer =
+        new ComposeContainer(new File(composeFile))
+            .withEnv("GRAVITINO_CI_DORIS_DOCKER_IMAGE", effectiveImage)
+            .withExposedService(
+                FE_SERVICE,
+                FE_MYSQL_PORT,
+                Wait.forListeningPort().withStartupTimeout(Duration.ofMinutes(3)))
+            .withExposedService(FE_SERVICE, FE_HTTP_PORT)
+            .withScaledService(BE_SERVICE, BE_SCALE)
+            .withStartupTimeout(Duration.ofMinutes(5))
```
Comment (yuqi1129) [inline integration-test-common/src/test/java/org/apache/gravitino/integration/test/container/DorisContainer.java]: Does it really take so long to start a Doris cluster?
